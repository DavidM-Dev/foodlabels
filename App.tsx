import React, { useState, useEffect } from 'react';
import { StyleSheet, Text, View, Button, TouchableOpacity } from 'react-native';
import { Camera } from 'expo-camera';
import * as Permissions from 'expo-permissions';

export default function App() {

  const [ hasCameraPermission, setCameraPermission ] = useState(null);
  const [ type, setType ] = useState(Camera.Constants.Type.back);

  useEffect(() => {
    Permissions.askAsync(Permissions.CAMERA).then(({ status }) => {
      setCameraPermission(status === 'granted');
    });
  }, []);

  return (
    <View style={{ flex: 1 }}>
      { hasCameraPermission && (
          <Camera style={{ flex: 1 }} type={type}>
            <View style={{
              flex: 1,
              backgroundColor: 'transparent',
              flexDirection: 'row',
              }}>
              <TouchableOpacity style={{
                flex: 0.1,
                alignSelf: 'flex-end',
                alignItems: 'center'
              }}
              onPress={() => {
                setType(
                    type === Camera.Constants.Type.back
                      ? Camera.Constants.Type.front
                      : Camera.Constants.Type.back,
                );
              }}>
                <Text style={{ fontSize: 18, marginBottom: 10, color: 'white' }}> Flip </Text>
              </TouchableOpacity>
            </View>
          </Camera>
      )}
      { !hasCameraPermission && (
        <Text>No camera permissions :(</Text>
      )}
    </View>
  );
}
