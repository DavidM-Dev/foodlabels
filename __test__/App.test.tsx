import React from 'react';
import renderer from 'react-test-renderer';

// @ts-ignore
import App from '../App.tsx';

describe('<App />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
