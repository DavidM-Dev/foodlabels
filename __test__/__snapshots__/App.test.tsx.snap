// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text>
    No camera permissions :(
  </Text>
</View>
`;
